echopype/echodata/echodata.py

```diff
@@ -312,6 +312,145 @@ def squeeze_non_scalar(n):
                 "this method only supports sonar_model values of AZFP, EK60, and EK80"
             )
 
+    def update_platform(self, extra_platform_data: xr.Dataset, time_dim="time"):
+        """
+        Updates the `EchoData.platform` group with additional external platform data.
+
+        `extra_platform_data` must be an xarray Dataset.
+        The name of the time dimension in `extra_platform_data` is specified by the
+        `time_dim` parameter.
+        Data is extracted from `extra_platform_data` by variable name; only the data
+        in `extra_platform_data` with the following variable names will be used:
+            - `"pitch"`
+            - `"roll"`
+            - `"heave"`
+            - `"latitude"`
+            - `"longitude"`
+            - `"water_level"`
+
+        Only `sonar_model`s of `"EK60"`, `"EK80"`, `"EA640"`, and `"AZFP"` are currently supported.
```

Comment:
@imranmaj @emiliom : Could you please remind me why we don't allow this for AD2CP files? I seem to remember that we discussed this but can't recall what the conversation was.

Reply:
I thinks that's for you and @imranmaj to answer. I don't remember being in a conversation about it, and wouldn't know if there's anything specific about AD2CP files that creates a special issue. I assume there isn't, and that its absence was just inertia.

Reply:
Inertia! 😅 

@imranmaj : do you see problems with adding AD2CP support in this? The potential "problem" I can see is that AD2CP files comes with `pitch`, `roll` and `heading` but not `lat` `lon`, so it is a partially filled dataset. So suppose users then want to add `lat` `lon`, the operations would become reading in the existing Platform group and add the new variables. Is this accurate?

Perhaps more generally the approach would be to:
- check if there is duplications in variables in `extra_platform_data` with the existing `ed.platform` dataset, warn user of overwritting if that's what they select
- add an option to allow user to specify if they want to overwrite or use existing variables
  - if allow overwriting then replace everything
- add new variables not already exist in `ed.platform`

What do you think?

Reply:
For what it's worth, the need you've outlined is more or less identical to the situation for the OOI AZFP data. The platform group is created with some variables, but empty lat & lon.

Reply:
Agreed!